OAuthSDK/YMLoginClient.m

```diff
@@ -155,69 +166,113 @@ - (BOOL)handleLoginRedirectFromUrl:(NSURL *)url sourceApplication:(NSString *)so
+                      
+                      // Retrieve all network tokens
+                      [self retrieveAllNetworkTokensWithToken:authToken completion:^(NSError *error) {
+                          if (!error) {
+                              [weakSelf.delegate loginClient:weakSelf didCompleteWithAuthToken:authToken];
+                              [[NSNotificationCenter defaultCenter] postNotificationName:YMYammerSDKLoginDidCompleteNotification
+                                                                                  object:weakSelf
+                                                                                userInfo:@{YMYammerSDKAuthTokenUserInfoKey: authToken}];
+                          } else {
+                              [self clearAuthToken];
+                              
+                              [weakSelf.delegate loginClient:weakSelf didFailWithError:error];
+                              [[NSNotificationCenter defaultCenter] postNotificationName:YMYammerSDKLoginDidFailNotification
+                                                                                  object:weakSelf
+                                                                                userInfo:@{YMYammerSDKErrorUserInfoKey: error}];
+                          }
+                      }];
+                  }
+                  failure:^(NSInteger statusCode, NSError *error) {
+                      NSMutableDictionary *userInfo = [@{NSLocalizedDescriptionKey: @"Unable to retrieve authentication token from code"} mutableCopy];
+                      
+                      if (error) {
+                          userInfo[NSUnderlyingErrorKey] = error;
+                          userInfo[NSLocalizedFailureReasonErrorKey] = [error localizedDescription];
+                      }
+                      
+                      NSError *newError = [NSError errorWithDomain:YMYammerSDKErrorDomain
+                                                              code:YMYammerSDKLoginObtainAuthTokenError
+                                                          userInfo:userInfo];
+                      
+                      [weakSelf.delegate loginClient:weakSelf didFailWithError:newError];
+                      [[NSNotificationCenter defaultCenter] postNotificationName:YMYammerSDKLoginDidFailNotification
+                                                                          object:weakSelf
+                                                                        userInfo:@{YMYammerSDKErrorUserInfoKey: newError}];
+                  }
      ];
 }
 
+- (void)refreshNetworkTokensWithCompletion:(void (^)(NSError *error))completion
+{
+    [self retrieveAllNetworkTokensWithToken:[self storedAuthToken] completion:completion];
+}
+
+- (void)retrieveAllNetworkTokensWithToken:(NSString *)authToken completion:(void (^)(NSError *error))completion
```

Comment:
I think the completion should be `void(^)(BOOL success, NSError *error)`. This way is cleaner I think.